apps/server/test/modules/task/task.e2e-spec.ts

```diff
@@ -1121,8 +1121,46 @@ describe('Task Controller (e2e)', () => {
 			expect(responseTask.courseId).toEqual(updateTaskParams.courseId);
 			expect(responseTask.lessonName).toEqual(lesson.name);
 		});
+		describe('business logic errors', () => {
+			it('POST should fail if NOT availableDate < dueDate', async () => {
```

Comment:
Maybe also check edge cases? Like equal or very close dates?

Reply:
comparing date objects, which coveres "very close dates". As for the dates to be equal, we don't have that business case yet. We can change that latter, but for now, why create a task, which is expired at the moment when is published?